src/app/public/modules/repeater/repeater.service.ts

```diff
@@ -3,15 +3,60 @@ import {
   Injectable
 } from '@angular/core';
 
+import {
+  BehaviorSubject
+} from 'rxjs/BehaviorSubject';
+
 import {
   SkyRepeaterItemComponent
 } from './repeater-item.component';
 
 @Injectable()
 export class SkyRepeaterService {
+
+  public activeItemId: BehaviorSubject<string> = new BehaviorSubject(undefined);
+
   public itemCollapseStateChange = new EventEmitter<SkyRepeaterItemComponent>();
 
-  public onItemCollapseStateChange(item: SkyRepeaterItemComponent) {
+  public items: Array<SkyRepeaterItemComponent> = new Array<SkyRepeaterItemComponent>();
+
+  public activateItemByIndex(index: number): void {
+    if (index === undefined) {
+      this.activeItemId.next(undefined);
+    } else {
+        const activeItem = this.items[index];
+        if (activeItem) {
+          this.activeItemId.next(activeItem.itemId);
+        }
+    }
+  }
+
+  public addItem(item: SkyRepeaterItemComponent): void {
+    this.items.push(item);
+  }
+
+  public destroyItem(item: SkyRepeaterItemComponent): void {
```

Comment:
Same here, we're not destroying the repeater item (in the Angular sense), so it might be better to rename this to something like `unregisterItem`.

Reply:
Done.